src/v3/src/transformer/layout/consent/transformEnduserConsent.ts

```diff
@@ -0,0 +1,151 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { doesI18NKeyExist } from '../../../../../v2/ion/i18nTransformer';
+import {
+  ButtonElement,
+  ButtonType,
+  ConsentScope,
+  DescriptionElement,
+  FieldElement,
+  IdxStepTransformer,
+  LinkElement,
+} from '../../../types';
+import { loc } from '../../../util';
+import { removeUIElementWithName } from '../../utils';
+
+export const transformEnduserConsent: IdxStepTransformer = ({ transaction, formBag }) => {
+  const { uischema, data } = formBag;
+  const {
+    // @ts-expect-error OKTA-598868 app is missing from rawIdxState type
+    rawIdxState: { app },
+    nextStep: {
+      name: stepName,
+      // @ts-expect-error OKTA-599472 scopes array missing from NextStep type
+      scopes,
+    } = {},
+  } = transaction;
+
+  uischema.elements = removeUIElementWithName('consent', uischema.elements);
+
+  const descriptionEle: DescriptionElement = {
+    type: 'Description',
+    noMargin: true,
+    contentType: 'subtitle',
+    options: {
+      content: loc('oie.consent.scopes.granular.description', 'login'),
+      dataSe: 'consent-description',
+    },
+  };
+  uischema.elements.push(descriptionEle);
+
+  if (Array.isArray(scopes)) {
+    uischema.elements.push(...(scopes?.map((scope: ConsentScope) => {
+      const labelI18nKey = `consent.scopes.${scope.name}.label`;
+      const descrI18nKey = `consent.scopes.${scope.name}.desc`;
+      const isPredefinedLabel = doesI18NKeyExist(labelI18nKey);
+      const label = isPredefinedLabel ? loc(labelI18nKey, 'login') : scope.label;
+      const description = doesI18NKeyExist(descrI18nKey)
+        ? loc(descrI18nKey, 'login')
+        : scope.desc;
+      const fieldName = `consent.scopes.${scope.name}`;
+      data[fieldName] = true;
+      return {
+        type: 'Field',
+        translations: [
+          {
+            name: 'label',
+            i18nKey: labelI18nKey,
+            value: label,
+            noTranslate: !isPredefinedLabel,
+          },
+          {
+            name: 'description',
+            i18nKey: descrI18nKey,
+            value: description,
+            noTranslate: !isPredefinedLabel,
+          },
+        ],
+        options: {
+          inputMeta: {
+            name: fieldName,
+            type: 'boolean',
+            mutable: false,
+          },
+        },
+      } as FieldElement;
+    })));
+  }
+
+  const termsOfServiceHref = app?.value?.termsOfService
+    && (app.value.termsOfService as Record<string, Record<string, unknown>>).href;
+  if (termsOfServiceHref) {
+    const tosLink: LinkElement = {
+      type: 'Link',
+      options: {
+        label: loc('consent.required.termsOfService', 'login'),
+        href: termsOfServiceHref as string,
```

Comment:
According to the casting above it seems like `termsOfServiceHref` should be of type `Record<string, unknown>` but we cast to `string` here. Is this correct?

Reply:
`app?.value?.termsOfService` is of type `Record<string, unknown>`, since we want to get the actual href, we have to cast it as `Record<string, Record<string, unknown>>` then point to the `href` property. 

This is what the data looks like and why we have to do this: 

```
"termsOfService": {
     "href": "https://example.com/tos"
 },
```


And this is the typecasting logic: 

```const termsOfServiceHref = app?.value?.termsOfService
    && (app.value.termsOfService as Record<string, Record<string, unknown>>).href;```
 
 So `termsOfServiceHref` is actually of type unknown.

Reply:
Discussed this, applying fix to type cast.